Add procedures for Debian/Ubuntu upstream repositories (#2962)
theforeman/foreman-documentation: guides/common/modules/proc_adding-upstream-repositories-for-deb.adoc

[id="adding-upstream-repositories-for-{DL}-{DL-major-context}_{context}"]
= Adding upstream repositories for {DL} {DL-major}

This example creates a product and repositories for {DL} {DL-major}.

.Prerequisites
* xref:Extracting_GPG_Public_Key_Fingerprints_from_Release_Files_{context}[Extract the GPG public keys] from the `Release` file.
* xref:Importing_a_Custom_GPG_Key_{context}[Import the content credentials] into {Project}.

.Procedure
. In the {ProjectWebUI}, navigate to *Content* > *Products*.
. Click *Create Product* to create a product named `{DL} {DL-major}`.

> For more information, see xref:Creating_a_Custom_Product_{context}[].

Reviewer: Lines 12 and 31 both talk about creating a product. I understand why you wouldn't want to include the xref in both steps. So how about including it as `.Additional resources` at the end of the procedure?

Author: Makes sense, applied.

[id="adding-upstream-repositories-for-{DL}-{DL-major-context}_{context}"]
= Adding upstream repositories for {DL} {DL-major}

> This example creates a product and repositories for {DL} {DL-major}.

Reviewer: For example procedures, you could also consider using example blocks: https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/ref_example-settings-for-ldap-connections.adoc?plain=1
Nothing says 'example' like proper markup for examples :) AsciiDoctor would ensure that it's clear this is just an example; you could focus on the explaining.

> This example creates a product and repositories for {DL} {DL-major}.

Author: I'm not a big fan of the rendered output: https://docs.theforeman.org/nightly/Installing_Server/index-katello.html#Example_Settings_for_LDAP_Connections_foreman

IMHO it's fine as is. It's basically a table of URLs and values that users want to find in the docs.

No change.

.Prerequisites

> * xref:Extracting_GPG_Public_Key_Fingerprints_from_Release_Files_{context}[Extract the GPG public keys] from the `Release` file

Author: Yes, because it's different and differently documentation for RPM and DEB packages.

> * xref:Importing_a_Custom_GPG_Key_{context}[Import the content credentials] into {Project}

Author: Applied.

.Procedure
. In the {ProjectWebUI}, navigate to *Content* > *Products*.
. Click *Create Product* to create a product named `{DL} {DL-major}`.
For more information, see xref:Creating_a_Custom_Product_{context}[].

> . On the *Repositories* tab, click *New Repository* to create three repositories of type `deb` as follows:

Reviewer 2: This is how I would have written it. But you can ignore my suggestion because the rest of the procedures are written exactly the same way. :-)

```suggestion
. On the *Repositories* tab, click *New Repository* to create three `deb` repositories with the following parameter values:
```

Author: Applied. But I kinda dislike "click X to create three repositories" when it's actually "click X to create one repo; redo twice". What do you think?

+
* *{DL} {DL-major} main*
** *Upstream URL*: `{DL-repo-main-url}`
** *Releases/Distributions*: `{DL-codename}`
** *Component*: `main`
** *Architecture*: `amd64`
* *{DL} {DL-major} updates*
** *Upstream URL*: `{DL-repo-updates-url}`
** *Releases/Distributions*: `{DL-codename}-updates`
** *Component*: `main`
** *Architecture*: `amd64`
* *{DL} {DL-major} security*
** *Upstream URL*: `{DL-repo-security-url}`
** *Releases/Distributions*: `{DL-codename}-security`
** *Component*: `main`
** *Architecture*: `amd64`
. Click *Create Product* to create a product named `{DL} {DL-major} client`.

> For more information, see xref:Creating_a_Custom_Product_{context}[].

Author: Dropped.
